isolate_example/lib/page_three.dart

```diff
@@ -67,6 +69,7 @@ class DataTransferIsolateController extends ChangeNotifier {
   SendPort _newIceSP;
 
   final currentProgress = <String>[];
+  List<bool> runningTest = [false, false, false];
```

Comment:
If you're only ever running one test at a time, have you considered simply storing the ID of the running test, rather than a list of bools? You could then zero out the "ID of the running test" when nothing was running and avoid keeping a separate `running` property.